ApeWorX/ape
# feat: subprocess provider mixin class

### What I did


fixes: #518 

### How I did it

Mostly moved logic from `ape-hardhat` to here; here is the matching `ape-hardhat` PR: https://github.com/ApeWorX/ape-hardhat/pull/30.

Now, plugin developers will need less code to create providers based on subprocesses. So we can probably create a `ganache` one easier now. I am using this elsewhere too, as you may know!

### How to verify it

I tested out both hardhat providers and and ran the `ape-hardhat` test suite locally.
You can do something similar! Try using both hardhat provider in a script or the console or tests. Make sure it works as before.

### Checklist
